sys/net/gnrc/link_layer/lwmac/include/lwmac_internal.h

```diff
@@ -98,186 +98,208 @@ typedef struct {
  *
- * @param[in] dev          ptr to netdev device
+ * @param[in] netif        ptr to the network interface
  *
  * @return                 true if active
  * @return                 false if not active
  */
-static inline bool gnrc_netdev_lwmac_get_dutycycle_active(gnrc_netdev_t *dev)
+static inline bool gnrc_lwmac_get_dutycycle_active(gnrc_netif2_t *netif)
 {
-    return (dev->lwmac.lwmac_info & GNRC_LWMAC_DUTYCYCLE_ACTIVE);
+    return (netif->mac.lwmac.lwmac_info & GNRC_LWMAC_DUTYCYCLE_ACTIVE);
 }
 
 /**
- * @brief set the needs-rescheduling flag of LWMAC
+ * @brief set the @ref GNRC_LWMAC_NEEDS_RESCHEDULE flag of LWMAC
  *
- * @param[in] dev          ptr to netdev device
- * @param[in] reschedule   value for LWMAC needs-rescheduling flag
+ * @param[in] netif        ptr to the network interface
+ * @param[in] reschedule   value for @ref GNRC_LWMAC_NEEDS_RESCHEDULE flag
  *
  */
-static inline void gnrc_netdev_lwmac_set_reschedule(gnrc_netdev_t *dev, bool reschedule)
+static inline void gnrc_lwmac_set_reschedule(gnrc_netif2_t *netif, bool reschedule)
 {
     if (reschedule) {
-        dev->lwmac.lwmac_info |= GNRC_LWMAC_NEEDS_RESCHEDULE;
+    	netif->mac.lwmac.lwmac_info |= GNRC_LWMAC_NEEDS_RESCHEDULE;
     }
     else {
-        dev->lwmac.lwmac_info &= ~GNRC_LWMAC_NEEDS_RESCHEDULE;
+    	netif->mac.lwmac.lwmac_info &= ~GNRC_LWMAC_NEEDS_RESCHEDULE;
     }
 }
 
 /**
- * @brief get the needs-rescheduling flag of LWMAC
+ * @brief get the @ref GNRC_LWMAC_NEEDS_RESCHEDULE flag of LWMAC
  *
- * @param[in] dev          ptr to netdev device
+ * @param[in] netif        ptr to the network interface
  *
  * @return                 true if needs rescheduling
  * @return                 false if no need for rescheduling
  */
-static inline bool gnrc_netdev_lwmac_get_reschedule(gnrc_netdev_t *dev)
+static inline bool gnrc_lwmac_get_reschedule(gnrc_netif2_t *netif)
 {
-    return (dev->lwmac.lwmac_info & GNRC_LWMAC_NEEDS_RESCHEDULE);
+    return (netif->mac.lwmac.lwmac_info & GNRC_LWMAC_NEEDS_RESCHEDULE);
 }
 
 /**
+ * @brief send a @ref net_gnrc_pkt "packet" over the network interface in LWMAC
+ *
+ * @pre `netif != NULL && pkt != NULL`
+ *
+ * @note The function re-formats the content of @p pkt to a format expected
+ *       by the netdev_driver_t::send() method of gnrc_netif_t::dev and
+ *       releases the packet before returning (so no additional release
+ *       should be required after calling this method).
+ *
+ * @param[in] netif The network interface.
+ * @param[in] pkt   A packet to send.
+ *
+ * @return  The number of bytes actually sent on success
+ * @return  -EBADMSG, if the @ref net_gnrc_netif_hdr in @p pkt is missing
+ *          or is in an unexpected format.
+ * @return  -ENOTSUP, if sending @p pkt in the given format isn't supported
+ *          (e.g. empty payload with Ethernet).
+ * @return  Any negative error code reported by gnrc_netif2_t::dev.
+ */
+int _gnrc_lwmac_transmit(gnrc_netif2_t *netif, gnrc_pktsnip_t *pkt);
```

Comment:
(optional) since the function name starts with a `_`: Maybe add `@internal` tag to the doc?

Reply:
How about I just change this function to `gnrc_lwmac_transmit()`??

Reply:
Added `@internal` tag to the doc.